File: P1/Ejercicio1/Bicicleta.java
package Ejercicio1;

/*
 * Bicicleta
 */
public abstract class Bicicleta {

    
    private int identificador;
    
    Bicicleta(int identificador){
        this.identificador=identificador;
    }
    public abstract Bicicleta crearBicicleta();
}

File: P1/P1/Ejercicio2/Bicicleta.py
from abc import ABC, abstractmethod
import copy

class Bicicleta(ABC):

    def __init__(self, identificador):
        self.identificador=identificador
    
    def __copy__(self):
        new = self.__class__(
            self.identificador
        )
        new.__dict__.update(self.__dict__)
        return new

    def __deepcopy__(self,memo=None):
        if memo is None:
            memo = {}

        
        new = self.__class__(
            self.identificador
        )
        new.__dict__= copy.deepcopy(self.__dict__,memo)
        
        return new



